From bcba04b55964d4802ee9b2e23150107b07f014be Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:18 -0400
Subject: [fs] nfsd: export proper maximum file size to the client

Message-id: <1403560314-32525-17-git-send-email-bfields@redhat.com>
Patchwork-id: 84607
O-Subject: [Patch RHEL7 016/172] nfsd: export proper maximum file size to the client
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Christoph Hellwig <hch@infradead.org>

I noticed that we export a way to high value for the maxfilesize
attribute when debugging a client issue.  The issue didn't turn
out to be related to it, but I think we should export it, so that
clients can limit what write sizes they accept before hitting
the server.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: aea240f4162d50e0f2d8bd5ea3ba11b5f072add8
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 1763b97..ee7237f 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -2392,7 +2392,7 @@ out_acl:
  if (bmval0 & FATTR4_WORD0_MAXFILESIZE) {
   if ((buflen -= 8) < 0)
    goto out_resource;
-  WRITE64(~(u64)0);
+  WRITE64(exp->ex_path.mnt->mnt_sb->s_maxbytes);
  }
  if (bmval0 & FATTR4_WORD0_MAXLINK) {
   if ((buflen -= 4) < 0)
-- 
1.7.1